engine/src/test/java/com/datatorrent/stram/engine/StreamingContainerTest.java

```diff
@@ -50,40 +56,66 @@ public void testCommitted() throws IOException, ClassNotFoundException
     String workingDir = new File("target/testCommitted").getAbsolutePath();
     lp.setAttribute(Context.OperatorContext.STORAGE_AGENT, new AsyncFSStorageAgent(workingDir, null));
     lp.setAttribute(DAGContext.CHECKPOINT_WINDOW_COUNT, 1);
-    CommitAwareOperator operator = lp.addOperator("CommitAwareOperator", new CommitAwareOperator());
+    String opName = "CommitAwareOperatorTestCommit";
+    CommitAwareOperator operator = lp.addOperator(opName, new CommitAwareOperator());
+    operator.setOpName(opName);
 
-    List<Long> myCommittedWindowIds = CommitAwareOperator.getCommittedWindowIdsContainer();
+    StramLocalCluster lc = new StramLocalCluster(lp);
+    lc.run(5000);
+
+    /* this is not foolproof but some insurance is better than nothing */
+    Assert.assertTrue("No Committed Windows", committedWindowIds.contains(opName));
+  }
+
+  @Test
+  public void testOiOCommitted() throws IOException, ClassNotFoundException
+  {
+    LogicalPlan lp = new LogicalPlan();
+    String workingDir = new File("target/testCommitted").getAbsolutePath();
+    lp.setAttribute(Context.OperatorContext.STORAGE_AGENT, new AsyncFSStorageAgent(workingDir, null));
+    lp.setAttribute(DAGContext.CHECKPOINT_WINDOW_COUNT, 1);
+    String op1Name = "CommitAwareInputOperatorTestOioCommit";
+    String op2Name = "CommitAwareProcessOperatorTestOioCommit";
+    CommitAwareInputOperator operator1 = lp.addOperator(op1Name, new CommitAwareInputOperator());
+    CommitAwareProcessOperator operator2 = lp.addOperator(op2Name, new CommitAwareProcessOperator());
+    operator1.setOpName(op1Name);
+    operator2.setOpName(op2Name);
+    lp.addStream("local", operator1.output, operator2.input).setLocality(Locality.THREAD_LOCAL);
 
     StramLocalCluster lc = new StramLocalCluster(lp);
     lc.run(5000);
 
     /* this is not foolproof but some insurance is better than nothing */
-    Assert.assertSame("Concurrent Use detected", myCommittedWindowIds, CommitAwareOperator.committedWindowIds);
-    Assert.assertFalse("No Committed Windows", myCommittedWindowIds.isEmpty());
+    Assert.assertTrue("No Committed Windows", committedWindowIds.contains(op1Name));
+    Assert.assertTrue("No Committed Windows", committedWindowIds.contains(op2Name));
   }
 
   private static class CommitAwareOperator extends BaseOperator implements CheckpointListener, InputOperator
   {
-    public static ArrayList<Long> committedWindowIds;
-    public static ArrayList<Long> checkpointedWindowIds = new ArrayList<Long>();
+    private String opName;
```

Comment:
transient leads to it not being serialized. When commit is called the the value is null(uninitialized) in the operator created from the byte stream, and this "null" value is added to the hashSet causing the assert will fail. name is already present in the parent class BaseOperator, had added a new variable opName to avoid confusion. Should I keep this as "name" as well?